pkg/docker/docker.go

```diff
@@ -196,18 +201,34 @@ func (d *Docker) ComposeIsAuthorized(compose *types2.Config) bool {
 }
 
 // StackDeploy deploy a stack
-func (d *Docker) StackDeploy(ctx context.Context, stackName string, composeFile []byte) (output string, err error) {
+func (d *Docker) StackDeploy(ctx context.Context, stackName string, composeFile []byte, configFile []byte) (output string, err error) {
 	cmd := func(cli *command.DockerCli) error {
 		// Write the compose file to a temporary file
-		tmp, err := ioutil.TempFile("", stackName)
+		compose, err := ioutil.TempFile("", stackName)
 		if err != nil {
 			return err
 		}
-		defer os.Remove(tmp.Name()) // clean up
-		if _, err := tmp.Write(composeFile); err != nil {
+		defer os.Remove(compose.Name()) // clean up
+		if _, err := compose.Write(composeFile); err != nil {
 			return err
 		}
-		deployOpt := stack.NewDeployOptions(stackName, tmp.Name(), true)
+
+		if configFile != nil {
+			log.Println("Using client configuration file")
+
+			// Read client configuration file from reader
+			cf := configfile.ConfigFile{
+				AuthConfigs: make(map[string]types.AuthConfig),
+			}
+			if err := json.NewDecoder(bytes.NewReader(configFile)).Decode(&cf); err != nil {
+				return err
+			}
+
+			// Update the cli with the new configuration
+			cli.SetConfigFile(&cf)
```

Comment:
Thx ... and just to be clear ... there is no way to accomplish this by using our own type that embeds the Docker type and augments the behavior? This feels like driving a 2-penny nail with sledgehammer...